reason-interaction.el

```diff
@@ -0,0 +1,208 @@
+
```

Comment:
Probably we need to add the header like in `reason-mode.el` in all files (or my emacs complains):

```
;;; reason-mode.el --- A major emacs mode for editing Reason (based on rust-mode) -*-lexical-binding: t-*-
  ;; Portions Copyright (c) 2015-present, Facebook, Inc. All rights reserved.
```